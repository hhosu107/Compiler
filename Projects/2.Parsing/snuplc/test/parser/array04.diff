24,25c24,25
<         := <INVALID>
<           [ @A        <array 3  of <array 3  of <int>>> ] <INVALID>
---
>         := <int>
>           [ @A        <array 3  of <array 3  of <int>>> ] <int>
31c31
<           [ @A        <array 3  of <array 3  of <int>>> ] <INVALID>
---
>           [ @A        <array 3  of <array 3  of <int>>> ] <int>
35c35,36
<           [ @A        <array 3  of <array 3  of <int>>> ] <array 3  of <array 3  of <int>>>
---
>           &() <ptr(4) to <array 3  of <array 3  of <int>>>>
>             [ @A        <array 3  of <array 3  of <int>>> ] <array 3  of <array 3  of <int>>>
49,50c50,51
<             := <INVALID>
<               [ %P        <ptr(4) to <array  of <array  of <int>>>> ] <INVALID>
---
>             := <int>
>               [ %P        <ptr(4) to <array  of <array  of <int>>>> ] <int>
68c69
<               < <int>
---
>               < <bool>
76c77
<                 < <int>
---
>                 < <bool>
81c82
<                   [ @A        <array 3  of <array 3  of <int>>> ] <INVALID>
---
>                   [ @A        <array 3  of <array 3  of <int>>> ] <int>
