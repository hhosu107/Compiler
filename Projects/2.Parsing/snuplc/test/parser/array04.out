parsing 'array04.mod'...
successfully parsed.
  AST:
    CAstScope: 'array04'
      symbol table:
        [[
          [ @A        <array 3  of <array 3  of <int>>>           ]
          [ *DIM(<ptr(4) to <NULL>>,<int>) --> <int>           ]
          [ *DOFS(<ptr(4) to <NULL>>) --> <int>           ]
          [ *ReadInt() --> <int>           ]
          [ *Set(<ptr(4) to <array  of <array  of <int>>>>,<int>) --> <NULL>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<int>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteStr(<ptr(4) to <array  of <char>>>) --> <NULL>           ]
          [ @i        <int>           ]
          [ *print() --> <NULL>           ]
        ]]
      statement list:
        call [ *WriteInt(<int>) --> <NULL> ] <NULL>
          11111111 <int>
        call [ *WriteLn() --> <NULL> ] <NULL>
        call [ *print() --> <NULL> ] <NULL>
        := <INVALID>
          [ @A        <array 3  of <array 3  of <int>>> ] <INVALID>
            0 <int>
            0 <int>
          2 <int>
        := <int>
          [ @i        <int> ] <int>
          [ @A        <array 3  of <array 3  of <int>>> ] <INVALID>
            0 <int>
            0 <int>
        call [ *Set(<ptr(4) to <array  of <array  of <int>>>>,<int>) --> <NULL> ] <NULL>
          [ @A        <array 3  of <array 3  of <int>>> ] <array 3  of <array 3  of <int>>>
          [ @i        <int> ] <int>
        call [ *WriteInt(<int>) --> <NULL> ] <NULL>
          22222222 <int>
        call [ *WriteLn() --> <NULL> ] <NULL>
        call [ *print() --> <NULL> ] <NULL>
      nested scopes:
        CAstScope: 'Set'
          symbol table:
            [[
              [ %P        <ptr(4) to <array  of <array  of <int>>>>               ]
              [ %v        <int>               ]
            ]]
          statement list:
            := <INVALID>
              [ %P        <ptr(4) to <array  of <array  of <int>>>> ] <INVALID>
                [ %v        <int> ] <int>
                [ %v        <int> ] <int>
              1 <int>
          nested scopes:
            empty.
        
        CAstScope: 'print'
          symbol table:
            [[
              [ $i        <int>               ]
              [ $j        <int>               ]
            ]]
          statement list:
            := <int>
              [ $i        <int> ] <int>
              0 <int>
            while cond
              < <int>
                [ $i        <int> ] <int>
                3 <int>
            while-body
              := <int>
                [ $j        <int> ] <int>
                0 <int>
              while cond
                < <int>
                  [ $j        <int> ] <int>
                  3 <int>
              while-body
                call [ *WriteInt(<int>) --> <NULL> ] <NULL>
                  [ @A        <array 3  of <array 3  of <int>>> ] <INVALID>
                    [ $i        <int> ] <int>
                    [ $j        <int> ] <int>
                := <int>
                  [ $j        <int> ] <int>
                  add <int>
                    [ $j        <int> ] <int>
                    1 <int>
              := <int>
                [ $i        <int> ] <int>
                add <int>
                  [ $i        <int> ] <int>
                  1 <int>
            call [ *WriteLn() --> <NULL> ] <NULL>
          nested scopes:
            empty.
        
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -oarray04.mod.ast.pdf array04.mod.ast.dot


Done.
